src/piggui.rs

```diff
@@ -34,19 +34,21 @@ fn main() -> Result<(), iced::Error> {
     })
 }
 
-struct Gpio {
+pub struct Gpio {
     // TODO this filename will be used when we add a SAVE button or similar
     #[allow(dead_code)]
     config_file: Option<String>, // filename where to load and save config file to/from
     gpio_description: [PinDescription; 40],
     gpio_config: GPIOConfig,
+    pub pin_function_selected: Vec<PinFunction>,
     clicked: bool,
 }
 
 #[derive(Debug, Clone, Copy)]
 
-enum Message {
+pub enum Message {
     Activate,
+    PinFunctionSelected(usize, PinFunction),
```

Comment:
For below, this might need to be Option<PinFunction> and None is when the pin has not been configured?

Reply:
Ok will check that out as well